From 4104c7099b0afc7e64505c9e0c61ec62c836bd26 Mon Sep 17 00:00:00 2001
From: Andreas Buhr <andreas.buhr@qt.io>
Date: Tue, 6 Oct 2020 15:07:06 +0200
Subject: [PATCH 049/171] Correct processEvents documentation

The documentation of processEvents states that it processes all
pending events. However, this is not true for the glib backend
and probably was not true for at least a decade.
QEventDispatcherGlib::processEvents(ProcessEventsFlags)
calls "g_main_context_iteration" which is documented
to only process events of highest priority:
https://developer.gnome.org/glib/2.66/glib-The-Main-Event-Loop.html#g-main-context-iteration

Fixes: QTBUG-87307
Task-number: QTBUG-84291
Change-Id: I82c6f9dba865bb07f981ad648a128316749682ca
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 98f4c97768494a86b1598562335f6cf36e4351d7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qcoreapplication.cpp | 4 ++--
 src/corelib/kernel/qeventloop.cpp       | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/corelib/kernel/qcoreapplication.cpp b/src/corelib/kernel/qcoreapplication.cpp
index 1bb63f80ac..f761696aea 100644
--- a/src/corelib/kernel/qcoreapplication.cpp
+++ b/src/corelib/kernel/qcoreapplication.cpp
@@ -1252,8 +1252,8 @@ bool QCoreApplication::closingDown()
 
 
 /*!
-    Processes all pending events for the calling thread according to
-    the specified \a flags until there are no more events to process.
+    Processes some pending events for the calling thread according to
+    the specified \a flags.
 
     You can call this function occasionally when your program is busy
     performing a long operation (e.g. copying a file).
diff --git a/src/corelib/kernel/qeventloop.cpp b/src/corelib/kernel/qeventloop.cpp
index 5a5dfb06aa..dc53af7018 100644
--- a/src/corelib/kernel/qeventloop.cpp
+++ b/src/corelib/kernel/qeventloop.cpp
@@ -118,8 +118,8 @@ QEventLoop::~QEventLoop()
 
 
 /*!
-    Processes pending events that match \a flags until there are no
-    more events to process. Returns \c true if pending events were handled;
+    Processes some pending events that match \a flags.
+    Returns \c true if pending events were handled;
     otherwise returns \c false.
 
     This function is especially useful if you have a long running
-- 
2.31.1

